hazelcast/include/hazelcast/client/internal/socket/BaseSocket.h

```diff
@@ -125,12 +125,14 @@ class BaseSocket : public hazelcast::client::socket
             add_invocation_to_map(connection, invocation, message);
 
             auto& datas = message->get_buffer();
-            std::vector<boost::asio::const_buffer> buffers;
-            buffers.reserve(datas.size());
+
+            struct outbox_entry_t outbox_entry;
+            outbox_entry.buffers.reserve(datas.size());
             for (const auto& data : datas) {
-                buffers.emplace_back(boost::asio::buffer(data));
+                outbox_entry.buffers.emplace_back(boost::asio::buffer(data));
             }
-            this->outbox_.push_back(buffers);
+            outbox_entry.invocation = invocation;
```

Comment:
If `invocation` parameter is removed from the `do_write` method, `invocation` becomes unnecessary after this line. So I suggest it to be moved. Btw, lambda object should be marked as `mutable` to make `move` operation valid.

```suggestion
            outbox_entry.invocation = std::move(invocation);
```

Reply:
fixed